azul-css-parser/src/css.rs

```diff
@@ -81,48 +82,83 @@ impl_display! { CssPseudoSelectorParseError<'a>, {
 
 fn pseudo_selector_from_str<'a>(data: &'a str) -> Result<CssPathPseudoSelector, CssPseudoSelectorParseError<'a>> {
     match data {
-        "first" => Ok(CssPathPseudoSelector::First),
-        "last" => Ok(CssPathPseudoSelector::Last),
-        "hover" => Ok(CssPathPseudoSelector::Hover),
-        "active" => Ok(CssPathPseudoSelector::Active),
-        "focus" => Ok(CssPathPseudoSelector::Focus),
-        other => {
-            // TODO: move this into a seperate function
-            if other.starts_with("nth-child") {
-                let mut nth_child = other.split("nth-child");
-                nth_child.next();
-                let mut nth_child_string = nth_child.next().ok_or(CssPseudoSelectorParseError::UnknownSelector(other))?;
-                nth_child_string.trim();
-                if !nth_child_string.starts_with("(") || !nth_child_string.ends_with(")") {
-                    return Err(CssPseudoSelectorParseError::UnclosedBracesNthChild(other));
-                }
+        "first" => Ok(First),
+        "last" => Ok(Last),
+        "hover" => Ok(Hover),
+        "active" => Ok(Active),
+        "focus" => Ok(Focus),
+        nth_child if data.starts_with("nth-child") => parse_nth_child_selector(nth_child),
+        other => Err(CssPseudoSelectorParseError::UnknownSelector(other)),
+    }
+}
 
-                // Should the string be empty, then the `starts_with` and `ends_with` won't succeed
-                let mut nth_child_string = &nth_child_string[1..nth_child_string.len() - 1];
-                nth_child_string.trim();
-                let parsed = nth_child_string.parse::<usize>()?;
-                Ok(CssPathPseudoSelector::NthChild(parsed))
-            } else {
-                Err(CssPseudoSelectorParseError::UnknownSelector(other))
-            }
-        },
+fn parse_nth_child_selector(input: &str) -> Result<CssPathPseudoSelector, CssPseudoSelectorParseError> {
+    let mut nth_child = input.split("nth-child");
+    nth_child.next();
+    let mut nth_child_string = nth_child.next().ok_or(CssPseudoSelectorParseError::UnknownSelector(input))?;
+    nth_child_string = nth_child_string.trim();
+    if !nth_child_string.starts_with("(") || !nth_child_string.ends_with(")") {
+        return Err(CssPseudoSelectorParseError::UnclosedBracesNthChild(input));

```

Comment:
Could have better erroring for open brace (I know it was like this before; still I think this could be changed)